lib/web/apiserver.go

```diff
@@ -197,6 +197,11 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*RewritingHandler, error) {
 	h.GET("/webapi/saml/sso", httplib.MakeHandler(h.samlSSO))
 	h.POST("/webapi/saml/login/console", httplib.MakeHandler(h.samlSSOConsole))
 
+	// Github connector handlers
+	h.GET("/webapi/github/login/web", httplib.MakeHandler(h.githubLoginWeb))
```

Comment:
if we have time, i suggest to use 1 endpoint for handling all types of oauth. web logins.
`/webapi/oauth?type=XXX&rest...`

It should remove duplicate code (which duplicates csrf checks) and make it easier to add a new type in the future (no need to modify web app code).

This is going to be a safe change as it's used only by web app.

Reply:
we don't have any other oauth logins at the moment, so this is not relevant.

Reply:
h.GET("/webapi/oidc/login/web", httplib.MakeHandler(h.oidcLoginWeb))
h.GET("/webapi/saml/sso", httplib.MakeHandler(h.samlSSO))
h.GET("/webapi/github/login/web", httplib.MakeHandler(h.githubLoginWeb))

all of them do the same thing for the most part. my suggestion is to use 1 endpoint and pass a provider type

Reply:
@alex-kovoy from Golang's URL router's perspective there's no difference between `/one/two` and `/one?arg=two`. you can route all 3 to the same function.

Reply:
@kontsevoy this is about code duplication. We have 3 functions doing the same thing for each above route. The suggestion is to have 1 function to reduce code duplication. 

We do not need to have 3 different routes either.

Reply:
there are similar, but not the same, so let's keep it as is for now.